src/sqlfluff/dialects/dialect_exasol.py

```diff
@@ -2332,13 +2332,13 @@ class FileOptionSegment(BaseSegment):
 ############################
 # USER
 ############################
-class CreateUserSegment(BaseSegment):
+class CreateUserStatementSegment(BaseSegment):
```

Comment:
Is it worth using inheritance for some of these segment types? Then you'd be inheriting the `type` property, as well as all the `is_...` properties.

Reply:
`CreateUser` doesn't exist in ANSI (maybe it should) but `CreateRole` does.

BTW this was one advantage if the `replace=True` decorator - was more obviously which ones were overriding and which weren't.

Reply:
We have a `Drop User` in ANSI so I added a `CREATE USER` too. Which allows this to be inherited now.

Reply:
Re: `replace`, what if we provided a `@new_segment` decorator that you **have** to use on new segments, and anything else is assumed to be (and automatically verified to be) a replacement?

Reply:
Not sure it's worth the effort. It's not a big deal. Was just a little handy to know that.